src/UiPath.Workflow/Activities/ExpressionCompiler.cs

```diff
@@ -35,6 +35,19 @@ protected static Assembly GetAssemblyForType(ITypeSymbol type)
         protected static Type GetSystemType(ITypeSymbol typeSymbol, Assembly assembly)
         {
             StringBuilder typeName = new();
+
+            if (typeSymbol is INamedTypeSymbol namedTypeSymbol)
+            {
+                if (namedTypeSymbol.IsGenericType)
+                {
+                    var arguments = namedTypeSymbol.TypeArguments.Select(t => GetSystemType(t, GetAssemblyForType(t))).ToArray();
+                    var fullyQualifiedName = $"{namedTypeSymbol.ContainingNamespace.ToDisplayString()}.{namedTypeSymbol.MetadataName}";
+                    var genericDefinitionType = assembly.GetType(fullyQualifiedName);
+
+                    return genericDefinitionType.MakeGenericType(arguments);
+                }
+            }
```

Comment:
This code is treated 10 lines below, in method `GetTypeName(ITypeSymbol typeSymbol)`. It does the same check as this, and properly identifies both arrays and generic arguments from any type.